engine/src/main/java/org/pentaho/di/trans/steps/multimerge/MultiMergeJoinData.java

```diff
@@ -0,0 +1,93 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2017 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.multimerge;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.PriorityQueue;
+
+import org.pentaho.di.core.RowSet;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.row.RowMetaInterface;
+import org.pentaho.di.trans.step.BaseStepData;
+import org.pentaho.di.trans.step.StepDataInterface;
+
+/**
+ * @author Biswapesh
+ * @since 24-nov-2005
+ *
+ */
+
+public class MultiMergeJoinData extends BaseStepData implements StepDataInterface {
+  public static class QueueEntry {
+    public Object[] row;
+    public int index;
+  }
+
+  public static class QueueComparator implements Comparator<QueueEntry> {
+    MultiMergeJoinData data;
+
+    QueueComparator( MultiMergeJoinData data ) {
+      this.data = data;
+    }
+
+    @Override
+    public int compare( QueueEntry a, QueueEntry b ) {
+      try {
+        int cmp =
+          data.metas[a.index].compare(
+            a.row, data.metas[b.index], b.row, data.keyNrs[a.index], data.keyNrs[b.index] );
+        return cmp > 0 ? 1 : cmp < 0 ? -1 : 0;
+      } catch ( KettleException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)